### DISCUSSION TITLE
Report Export

### DISCUSSION THREAD

**Discussion by DominikRoB** [solidtime-io/solidtime#240] [Feature requests]
Coming from toggl, the feature I'm missing for a complete switch is the ability to export reports (of all entries for a time frame) to .csv files (or other machine readable formats).

Otherwise, or rather in general, solidtime is a great project, keep up the good work!

**Comment by Onatcer** [comment]
thanks for the feedback! Exportable and Sharable Reports are currently in progress. They will be one of the next features we will release. I'll leave this issue open and give an update once we have it ready.

**Comment by korridor** [comment]
We decided to move feature requests into github discussions, to separate feature requests from bugs and allow for better prioritization.

This feature is already available in solidtime cloud and in the `main` branch. It'll be available with the next release for everyone.

**Comment by em19** [comment]
After setting up a self-hosted instance with local storage driver, the download of the reports failed with the error "the driver doesn't allow to create temporary urls". I managed to fix it by manually adding `'serve' => true` to the `local` entry to config/filesystems.php within the app container.

**Reply by developersteve** [reply]
> After setting up a self-hosted instance with local storage driver, the download of the reports failed with the error "the driver doesn't allow to create temporary urls". I managed to fix it by manually adding `'serve' => true` to the `local` entry to config/filesystems.php within the app container.

Thank you, this worked for me too for a docker deployment.

**Reply by korridor** [reply]
Thanks for the feedback! This is fixed in the recently released version [0.5.0](https://github.com/solidtime-io/solidtime/releases/tag/v0.5.0).

**Comment by korridor** [comment]
✅ Since the version [0.4.0](https://github.com/solidtime-io/solidtime/releases/tag/v0.4.0) it is possible to reports in machine readable format like CSV. In the recently released version [0.5.0](https://github.com/solidtime-io/solidtime/releases/tag/v0.5.0) we also added PDF exports.
